Feature/separated volume change

DianaYu44 opened the pull request:
Added:
- Look-up table potential.
- Separated volume change.

DianaYu44 committed:
Volume move for rhombic boundary and correction to test

DianaYu44 committed:
Merge branch 'main' of https://github.com/ElkePahl/ParallelTemperingMonteCarlo.jl

DianaYu44 committed:
Merge branch 'main' of https://github.com/ElkePahl/ParallelTemperingMonteCarlo.jl

DianaYu44 committed:
r_cut

DianaYu44 committed:
Added look-up table potential

DianaYu44 committed:
changed the long range coefficient for look-up table

DianaYu44 committed:
Added documentation

DianaYu44 committed:
added volume change xy and z functions

DianaYu44 committed:
look up table

DianaYu44 committed:
Added look-up table test

DianaYu44 committed:
Try putting the test in runtest

DianaYu44 committed:
show path

DianaYu44 committed:
Try new path

DianaYu44 committed:
Move to test_potentials

DianaYu44 committed:
testing threads

DianaYu44 committed:
changes for energy update in magnetic field

DianaYu44 committed:
separated volume change, for B = 0 and B != 0

DianaYu44 committed:
Added rhombic box 150 atoms script

DianaYu44 committed:
Separated volume change for look-up table potential

DianaYu44 committed:
Small mistakes corrected

DianaYu44 committed:
correction

ElkePahl committed:
Merge branch 'main' into feature/separated_volume_change

DianaYu44 committed:
tests corrected

DianaYu44 committed:
Merge branch 'feature/separated_volume_change' of https://github.com/ElkePahl/ParallelTemperingMonteCarlo.jl into feature/separated_volume_change

DianaYu44 committed:
correction

DianaYu44 committed:
correction

DianaYu44 committed:
new tests about volume changes

DianaYu44 committed:
change

DianaYu44 committed:
rhombic potential b energy

DianaYu44 committed:
potential variables tan_mat

DianaYu44 committed:
separated volume change tests

DianaYu44 committed:
...

DianaYu44 committed:
metropolis npt test

ElkePahl commented on src/EnergyEvaluation.jl:
```diff
@@ -727,13 +943,37 @@ function energy_update!(trial_pos,index,config::Config,potential_variables::ELJP
+
     return potential_variables,new_en
 end
+
```
It would make sense to combine these two methods that just differ by using rcut or not (same for dimer_enery_update) - is that possible? e.g. by making  rcut a variable with a default value

ElkePahl commented on src/EnergyEvaluation.jl:
```diff
@@ -807,7 +1049,12 @@ end
     tan_matrix = get_tantheta_mat(config,config.bc)
 
-    return ELJPotentialBVariables{T}(zeros(N),tan_matrix,zeros(N))
+    return ELJPotentialBVariables{T}(zeros(N),tan_matrix,tan_matrix,zeros(N))
```
? why is tan_matrix there twice?

ElkePahl commented on src/EnergyEvaluation.jl:
```diff
@@ -807,7 +1049,12 @@ end
+function set_variables(config::Config{N,BC,T},dist2_matrix::Matrix,pot::LookuptablePotential) where {N,BC,T}
+    tan_matrix = get_tantheta_mat(config,config.bc)
+
+    return LookupTableVariables{T}(zeros(N),tan_matrix,tan_matrix,zeros(N))
```
same here?

ElkePahl commented on src/Exchange.jl:
```diff
@@ -56,6 +56,10 @@ Function returning the probability value associated with a trial move. Four meth
 """
 function metropolis_condition(delta_energy, beta)
     prob_val = exp(-delta_energy*beta)
+
```
please remove the commented out lines (also later in this file)

ElkePahl commented on src/Initialization.jl:
```diff
@@ -54,6 +54,8 @@ function initialisation(mc_params::MCParams,temp::TempGrid,start_config::Config,
     
     mc_states = [MCState(temp.t_grid[i], temp.beta_grid[i],start_config,ensemble,potential) for i in 1:mc_params.n_traj]
 
+    println(mc_states[1].en_tot)
```
do you want to always print that out?

ElkePahl commented on src/MCMoves.jl:
```diff
@@ -43,14 +45,14 @@ end
 function atom_displacement(pos, max_displacement, bc::CubicBC)
     delta_move = SVector((rand()-0.5)*max_displacement,(rand()-0.5)*max_displacement,(rand()-0.5)*max_displacement)
     trial_pos = pos + delta_move
```
please update documentation above

ElkePahl commented on src/MCMoves.jl:
```diff
@@ -107,13 +109,95 @@ function volume_change(conf::Config, bc::RhombicBC, max_vchange, max_length)
+    return new_pos
+end
+
+function volume_change_xy(conf::Config, bc::RhombicBC, max_vchange, max_length, lh_ratio)
```
documentation needed

ElkePahl commented on src/MCRun.jl:
```diff
@@ -89,7 +90,9 @@ end
 """
 function mc_step!(mc_states::Vector{stype},move_strat,pot,ensemble,n_steps) where stype <: MCState
-    for state in mc_states
+
```
add a comment here to explain threading

ElkePahl commented on src/MCStates.jl:
```diff
@@ -63,14 +67,24 @@ function max_length(bc::RhombicBC)
     return bc.box_length*1.8
 end
 
+function max_height(bc::SphericalBC)
```
documentation needed

ElkePahl commented on src/swap_config.jl:
```diff
@@ -54,13 +54,56 @@ function swap_config_v!(mc_state::MCState,bc::CubicBC,trial_config::Config,new_d
+
+    mc_state.ensemble_variables.r_cut = mc_state.ensemble_variables.new_r_cut
+end
+
```
documentation missing

ElkePahl reviewed (commented):
Could you please work on the comments; mainly documentation issues.

DianaYu44 committed:
update rdf for NPT ensemble

DianaYu44 committed:
correction to volume change moves and swap config

DianaYu44 committed:
threads

DianaYu44 committed:
removed checkpoint test for now

DianaYu44 committed:
crossed out checkpoint test

DianaYu44 committed:
different starting configs

DianaYu44 committed:
rectangular box

DianaYu44 committed:
change to rectangular rdf

DianaYu44 committed:
Update Sampling.jl

DianaYu44 committed:
some changes

DianaYu44 committed:
correction in dimer_energy

DianaYu44 committed:
try to make get_tan only return postive results by using abs

DianaYu44 committed:
new correction

DianaYu44 committed:
changed the first print out to 1000 cycles, reduce time to debug

DianaYu44 committed:
writeout en_tot from dimer energy config

DianaYu44 committed:
changed the print command

DianaYu44 committed:
changed the tan_mat update in swap_config.jl

DianaYu44 committed:
try volume_change in all directions only

DianaYu44 committed:
add xy-direction

DianaYu44 committed:
changed the tan_mat update in volume_change_xyz

DianaYu44 committed:
correction

DianaYu44 committed:
changed dimer_energy_config

DianaYu44 committed:
try changing dimer_energy_fig's arguments

DianaYu44 committed:
added types to tan_mat in dimer_energy_config

DianaYu44 committed:
update mcrun.jl

DianaYu44 committed:
printout mc_state.new_en

DianaYu44 committed:
restore the old energy evaluation

DianaYu44 committed:
correction

DianaYu44 committed:
try adding a update energy or not argument to potential_variables for potential in B

DianaYu44 committed:
try writing old results of swap_config_v!

DianaYu44 committed:
check volume_change_xyz

DianaYu44 committed:
...

DianaYu44 committed:
...

DianaYu44 committed:
...

DianaYu44 committed:
...

DianaYu44 committed:
.

DianaYu44 committed:
try get new_tan_mat directly from get_tantheta_mat function

DianaYu44 committed:
check new_tan_mat update

DianaYu44 committed:
.

DianaYu44 committed:
try copy new tan mat to tan mat

DianaYu44 committed:
..

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
try write out energy after each function

DianaYu44 committed:
check generate move function

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
...

DianaYu44 committed:
...

DianaYu44 committed:
.

DianaYu44 committed:
..

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
..

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
..

DianaYu44 committed:
try changing volume_change_xy and z to all directions

DianaYu44 committed:
add the new_tan_mat calculation

DianaYu44 committed:
changed dimer_energy_config

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
..

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
keep only volume moves

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
.

DianaYu44 committed:
different method in swap_config_v! for mc_state.p_v.tan_mat

DianaYu44 committed:
check

DianaYu44 committed:
.

DianaYu44 committed:
have a look at dimer_energy_config when new_tan_mat and tan_mat are different or the same

DianaYu44 committed:
print new_tan_mat and tan_mat in dimer_energy_config

DianaYu44 committed:
change conditon in of using tan_mat or new_tan_mat in dimer_energy_config

DianaYu44 committed:
e-9 to 10^(-9)

DianaYu44 committed:
acc_test! change, if not accepted, new_tan_mat is reset to tan_mat

DianaYu44 committed:
cross out the print lines in dimer_energy_config

DianaYu44 committed:
change mc_move! back to both atom and volume

DianaYu44 committed:
try changing functions for rectangular boundary

DianaYu44 committed:
changed the print command in mc_cycle!

DianaYu44 committed:
correction to the previous one

DianaYu44 committed:
change write out configuration back to every 10000 cycles

DianaYu44 committed:
delete the print energy commands in mc_cycle!

DianaYu44 committed:
Update MCRun.jl

DianaYu44 committed:
Update MCRun.jl

DianaYu44 committed:
Update MCRun.jl

mtsch committed:
Merge branch 'main' into feature/separated_volume_change

mtsch committed:
Make tests pass, reformatting

mtsch committed:
Merge branch 'main' into feature/separated_volume_change

mtsch committed:
Fix some scripts, add them to the tests

mtsch committed:
Remove double definition of ELJPotentialEven

mtsch committed:
Add cu38 to tests

mtsch committed:
split ne13 into normal and lookup versions

mtsch committed:
LookuptablePotential -> LookupTablePotential

mtsch committed:
Some notes and formatting

mtsch committed:
saving for LookupTablePotential

mtsch committed:
Remove plots from main directory

mtsch committed:
add some more tests and comments

mtsch committed:
Fix pbc32.jl, formatting

mtsch committed:
Merge branch 'main' into feature/separated_volume_change

mtsch committed:
Script fixes, minor edits in code.

mtsch committed:
Remove unused deps, add version bounds

mtsch committed:
Fix problem in MCRun if `save=true`

ElkePahl committed:
Delete scripts/energy_of_saved_configs.jl

Script is on analyzing configs and doesn't belong here. I stored it in the PTMC Dropbox folder under Diana/Removed_PTMC_Code, just in case.

ElkePahl committed:
Delete scripts/find_avg_ev.jl

calculates averages from histogram data - stored in Dropbox folder - if valuable we could change to a function, but at the moment it is quite specific

ElkePahl committed:
Update BoundaryConditions.jl 

updated documentation
added TODOs

ElkePahl committed:
Update BoundaryConditions.jl

changed documentation

ElkePahl committed:
Update BoundaryConditions.jl

changed latex expressions

ElkePahl committed:
Update BoundaryConditions.jl

ElkePahl committed:
Update BoundaryConditions.jl

ElkePahl committed:
Update BoundaryConditions.jl

ElkePahl committed:
Update BoundaryConditions.jl

ElkePahl committed:
Update BoundaryConditions.jl

ElkePahl committed:
Update BoundaryConditions.jl

ElkePahl committed:
Update BoundaryConditions.jl

ElkePahl committed:
Update BoundaryConditions.jl

ElkePahl committed:
Update Exchange.jl

removed commented out print statements

ElkePahl committed:
Update EnergyEvaluation.jl

added TODOs

ElkePahl committed:
Update MCRun.jl

changed some documentation strings - still lots left to be done

ElkePahl reviewed (approved):
Looks all good to me; fingers crossed
